iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/function_ref.inl

```diff
@@ -32,16 +51,22 @@ inline function_ref<ReturnType(ArgTypes...)>::function_ref(CallableType&& callab
 
 template <class ReturnType, class... ArgTypes>
 inline function_ref<ReturnType(ArgTypes...)>::function_ref(ReturnType (&function)(ArgTypes...)) noexcept
-{
     // the cast is required to work on POSIX systems
-    m_pointerToCallable = reinterpret_cast<void*>(function);
-
+    // AXIVION Next Construct AutosarC++19_03-A5.2.4, AutosarC++19_03-A5.2.4-M5.2.6 : Type-safety ensured by casting
+    // back on call NOLINTNEXTLINE (cppcoreguidelines-pro-type-reinterpret-cast)
+    : m_pointerToCallable(reinterpret_cast<void*>(function))
+    ,
     // the lambda does not capture and is thus convertible to a function pointer
     // (required by the C++ standard)
-    m_functionPointer = [](void* target, ArgTypes... args) -> ReturnType {
-        auto f = reinterpret_cast<ReturnType (*)(ArgTypes...)>(target);
+    m_functionPointer([](void* target, ArgTypes... args) -> ReturnType {
+        using PointerType = ReturnType (*)(ArgTypes...);
+        // AXIVION Next Construct AutosarC++19_03-A5.2.4 : Type-safety ensured by casting from type
```

Comment:
Type safety is ensured by the fact we only pass functions with this type as `target`, i.e. they are convertible.

This is entirely under the control of the class.

Reply:
@mossmaurice I still find the justification lacking here, hence my comment about the reason.

Without extra knowledge of how it works, I could not infer it from the justification.

More precisely
`the class design ensures a cast to the actual type of target`